provider/machine_pool_resource.go

```diff
@@ -75,6 +80,22 @@ func (t *MachinePoolResourceType) GetSchema(ctx context.Context) (result tfsdk.S
 				Type:        types.Int64Type,
 				Optional:    true,
 			},
+			"use_spot_instances": {
+				Description: "Use Spot Instances. Applicable only for AWS.",
+				Type:        types.BoolType,
+				Optional:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					ValueCannotBeChangedModifier(t.logger),
+				},
+			},
+			"max_spot_price": {
+				Description: "Max Spot price. Applicable only for AWS.",
```

Comment:
Since the scope is only AWS, we can omit `Applicable only for AWS.`

Reply:
updated variable description to remove `Applicable only for AWS.`